### PR TITLE
chore: increase unit test coverage, add some doctest

### PR DESCRIPTION
**Pull request by milmazz**

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -1,30 +0,0 @@
-# This file is responsible for configuring your application
-# and its dependencies with the aid of the Mix.Config module.
-use Mix.Config
```
Comment (milmazz) [inline config/config.exs:3]: Not needed

File: lib/circular_buffer.ex [hunk 2]
```diff
@@ -55,6 +60,16 @@ defmodule CircularBuffer do
 
   @doc """
   Returns the newest element in the buffer
+
+  ## Examples
+
+      iex> cb = CircularBuffer.new(3)
+      iex> CircularBuffer.newest(cb)
+      nil
```
Comment (milmazz) [inline lib/circular_buffer.ex:68]: This doctest was added mostly to cover the empty case.

File: lib/circular_buffer.ex [hunk 3]
```diff
@@ -69,6 +84,17 @@ defmodule CircularBuffer do
 
   @doc """
   Checks the buffer to see if its empty
+
+  Returns `true` if the given circular buffer is empty, otherwise `false`.
```
Comment (milmazz) [inline lib/circular_buffer.ex:88]: This function wasn't covered via `test/circular_buffer_test.exs`. So, I took the chance to add some docs here.

File: test/circular_buffer_test.exs [hunk 4]
```diff
@@ -148,16 +150,17 @@ defmodule CircularBufferTest do
 
   def size_and_list do
     let size <- pos_integer() do
-      let is <- ints(size*2, []) do
+      let is <- ints(size * 2 + 1, []) do
```
Comment (milmazz) [inline test/circular_buffer_test.exs:153]: With this "trick" we can cover:

```elixir
def oldest(%CB{b: [oldest | _rest]}), do: oldest
```

**Comment by keathley** [comment]
Thanks @milmazz

**Pull request merged**